lib/Models/RegionProvider.js

```diff
@@ -0,0 +1,405 @@
+     * @type {String[]}
+     */
+    this.aliases = defaultValue(properties.aliases, [this.regiontype]);
+    this.serverReplacements = properties.serverReplacements;
+    this.dataReplacements = properties.dataReplacements;
+    this.disambigColumns = properties.disambigColumns;
+    /** 
+     * Whether this region type uses text codes, rather than numeric. It matters because numeric codes are treated differently by the
+     * CSV handling models.
+     * @type {Boolean}
+     */
+    this.textCode = defaultValue(properties.textCodes, false); // yes, it's singular...
+
+    /**
+     * Array of attributes of each region, once retrieved from the server.
+     * @type {Object[]}
+     */
+    this.regions = [];
+
+
+};
+
+/**
+ * Turn a WFS list of attribute values into a straight array
+ * @param {String} propertyName: optional, for setting an additional (disambiguating) property on a region
+ *  Not sure if this is really necessary - we can surely access the name of that column when needed.
+ *
+ */
+RegionProvider.prototype.loadRegionsFromXML = function(xml, propertyName) {
+    var obj = xml2json(xml);
+
+    if (!defined(obj.member)) {
+        console.log(xml);
+        throw new ModelError("Couldn't load region boundaries for region " + this.regionProp);
+    }
+    
+    if (!(obj.member instanceof Array)) {
+        obj.member = [obj.member];
+    }
+    if (obj.member.length === 0) {
+        throw new ModelError("Zero region boundaries found for region " + this.regionProp);
+    }
+    for (var i = 0; i < obj.member.length; i++) {
+        if (this.regions[i] === undefined) {
+            this.regions[i] = {};
+        }
+        // either { id: 123 } or { id: 'Bunbury', STE_NAME: 'VIC' }
+        if (!propertyName) {
+            this.regions[i].id = obj.member[i][this.regionProp];
+        } else {
+            this.regions[i][propertyName] = obj.member[i][propertyName];
+        } 
+    }
+};
+
+/** 
+ * Given an entry from the region mapping config, contact our WFS server to get the list of all IDs of that type 
+ * @returns {Object} Promise that returns true if IDs were loaded for the first time, or false if already loaded.
+*/
+
+RegionProvider.prototype.loadRegionIDs = function() {
+    if (this.regions.length > 0) {
+        return when(false); // already loaded, so return insta-promise.
+    }
+    if (this.server === undefined) {
+        throw (new DeveloperError('No server for region mapping defined: ' + this.regionType));
+    }            
+    var that = this;
+
+    var baseuri = URI(this.server).addQuery({ 
+        service: 'wfs',
+        version: '2.0',
+        request: 'getPropertyValue',
+        typenames: this.layerName});
+
+    // get the list of IDs that we will attempt to match against for this column
+    var p = [];
+    var url = baseuri.addSearch('valueReference', this.regionProp).toString();
+    p.push(loadText(corsProxy.getURL(url))
```

Comment:
> My short term fix is to never proxy - our region mapping server(s) support CORS.

This will break region mapping in IE9, because IE9 doesn't support CORS at all.